apps/wallet/src/background/connections/index.ts

```diff
@@ -19,21 +28,46 @@ export class Connections {
                 switch (port.name) {
                     case ContentScriptConnection.CHANNEL:
                         connection = new ContentScriptConnection(port);
+                        this.#totalCsConnections++;
+                        this.#events.emit(
+                            'totalCsChanged',
+                            this.#totalCsConnections
+                        );
                         break;
                     case UiConnection.CHANNEL:
                         connection = new UiConnection(port);
+                        this.#totalUiConnections++;
+                        this.#events.emit(
+                            'totalUiChanged',
+                            this.#totalUiConnections
+                        );
                         break;
                     default:
                         throw new Error(
                             `[Connections] Unknown connection ${port.name}`
                         );
                 }
-                this._connections.push(connection);
+                this.#connections.push(connection);
                 connection.onDisconnect.subscribe(() => {
                     const connectionIndex =
-                        this._connections.indexOf(connection);
+                        this.#connections.indexOf(connection);
                     if (connectionIndex >= 0) {
-                        this._connections.splice(connectionIndex, 1);
+                        this.#connections.splice(connectionIndex, 1);
+                        if (connection instanceof UiConnection) {
+                            this.#totalUiConnections--;
```

Comment:
I don't love this bookkeeping. Would rather this either be not needed at all (most ideally) or just be computed out of the `#connections` themselves.

Reply:
Makes sense, chose to do this since we had already the events but I can change it